pkg/authenticator/api-key.go

```diff
@@ -54,20 +54,26 @@ func (aa *APIKeyAuthenticator) GetToken(freshTokenRequired bool) (string, uint64
 		}
 	}
 
-	aa.logger.Info("Fetching fresh token")
-	tokenResponse, err := aa.authenticator.RequestToken()
+	var tokenResponse *core.IamTokenServerResponse
+	err = retry(aa.logger, func() error {
+		tokenResponse, err = aa.authenticator.RequestToken()
+		return err
+	})
+
 	if err != nil {
-		aa.logger.Error("Error fetching fresh token", zap.Error(err))
-		// If the cluster cannot access private iam endpoint, hence returns timeout error, switch to public IAM endpoint.
-		if !isTimeout(err) {
+		// Reset the IAM URL to public, if it is private
+		if !resetIAMURL(aa) {
 			return "", tokenlifetime, utils.Error{Description: "Error fetching iam token using api key", BackendError: err.Error()}
 		}
-
-		aa.logger.Info("Updating iam URL to public, if it is private and retrying to fetch token")
-		if !resetIAMURL(aa) {
+		// Retry fetching IAM token again
+		aa.logger.Info("Updated IAM URL from private to public, retrying to fetch IAM token")
```

Comment:
can we rewrite this error message? 
retrying token fetch, after switching from private to public IAM URL..

Reply:
Done